Refactor : move current year and week from study to weekly optimization problem

guilpier-code opened the pull request:
The purpose is to move the information of the current year and week from the Study to the weekly optimization problem.
Why ?
- Because weekly problem is more the location of current year and week (the problem knows what year and week it's solving)
- It makes the code cleaner (reduces dependency on the study)
- It makes easier some changes to come (export MPS files when optimization is daily)

Some collateral cleanings were also made.

guilpier-code committed:
Moving current year and week from study to weekly optimization problem

Why ?
- Because weekly problem is more the location of current year and week (the problem knows what year and week it's solving)
- It makes the code cleaner (reduces dependency on the study)
- It makes easier some changes to come (export MPS files when optimization is daily)

Some collateral cleaning was made

guilpier-code committed:
move year and week from study to weekly opt problem : correction after sonar Q review

flomnes commented on src/solver/utils/mps_utils.cpp:
```diff
@@ -320,49 +324,60 @@ void OPT_EcrireJeuDeDonneesLineaireAuFormatMPS(void* Prob, int optNumber, uint n
 }
 void fullOrToolsMPSwriter::runIfNeeded(Solver::IResultWriter::Ptr writer)
 {
     // Make or-tools print the MPS files leads to a crash !
```
This comment is now irrelevant, please remove it

guilpier-code replied:
Done
**Question** : from which version of or-tools was the problem fixed ?

guilpier-code replied:
Done

flomnes commented on src/solver/utils/mps_utils.h:
```diff
@@ -28,39 +28,52 @@ void OPT_EcrireJeuDeDonneesLineaireAuFormatMPS(void*,
+       year_(year), week_(week), current_optim_number_(currentOptimNumber)
+    {}
+    I_MPS_writer()
+    {}
```
```suggestion
    I_MPS_writer(int year, int week, int currentOptimNumber) : 
       year_(year), week_(week), current_optim_number_(currentOptimNumber) = default;
    I_MPS_writer() = default;
```

guilpier-code replied:
Having a **default** constructor with an initialization list does not compile : 
```C++
I_MPS_writer(int year, int week, int currentOptimNumber) : 
       year_(year), week_(week), current_optim_number_(currentOptimNumber) = default;
```

guilpier-code replied:
**default** identifier after initialization list does not compile

flomnes commented on src/solver/simulation/sim_structure_probleme_economique.h:
```diff
@@ -490,6 +490,10 @@ struct AdequacyPatchParameters
 
 struct PROBLEME_HEBDO
 {
+    int dayInTheYear = 0;
+    int weekInTheYear = 0;
+    int year = 0;
```
`unsigned int`

guilpier-code replied:
Done

guilpier-code replied:
done

guilpier-code committed:
move year and week from study to weekly opt problem : correction after human review

flomnes commented on src/solver/simulation/sim_structure_probleme_economique.h:
```diff
@@ -490,6 +490,10 @@ struct AdequacyPatchParameters
 
 struct PROBLEME_HEBDO
 {
+    uint dayInTheYear = 0;
```
Unused

guilpier-code replied:
It is removed in #1041

guilpier-code committed:
Refactor : about optimization number (#1040)

* Refactor : about optimization number

- Information of optimization number is spread at 2 locations : it can be found inside ProblemeHebdo and inside ProblemeHebdo->ProblemeAResoudre
- This piece of information does not need to be nested in these structures
So it is declared and defined once, and locally, and then passed to lower functions when needed.

* Refactor : about optimization number : oops, correction of a unexpected remove

* Refactor : about optimization number : removing unnecessary trailing affectation

* [skip ci] Refactor about optimization number : rename the optimization number from french to english
